charts/fluent-bit/templates/daemonset.yaml

```diff
@@ -26,17 +26,23 @@ spec:
   {{- end }}
   template:
     metadata:
+      labels:
+        {{- include "fluent-bit.selectorLabels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    {{- if or (not .Values.hotReload.enabled) .Values.podAnnotations }}
       annotations:
+      {{- if not .Values.hotReload.enabled }}
```

Comment:
Ah good, was going to say with hot reload the checksum annotation trick to trigger rollout is not needed.